src/AndcultureCode.CSharp.Sitefinity.Conductors/Domain/DynamicContentRepositoryConductor.cs

```diff
@@ -61,6 +68,11 @@ ILogger<IDynamicContentRepositoryConductor> logger
             var dynamicModuleManager = DynamicModuleManager.GetManager(DefaultProviderName);
             dynamicModuleManager.SaveChanges();
 
+            if (publish)
+            {
+                dynamicModuleManager.Lifecycle.Publish(dataItem);
```

Comment:
I believe we should be setting the `ApprovalWorkflowState` of `dataItem` to `ApprovalWorkflowStates.PUBLISHED`, right?

Additionally, if it's published, shouldn't we be returning the published item? `dynamicModuleManager.Lifecycle.Publish(dataItem)` should return the newly published item.

Reply:
Yea, I was wondering if it made sense to return the published data item. Lemme make those changes quick!